zebra-consensus/src/transaction.rs

```diff
@@ -90,20 +93,26 @@ pub enum Request {
 }
 
 impl Request {
+    /// The transaction to verify that's in this request.
     pub fn transaction(&self) -> Arc<Transaction> {
         match self {
             Request::Block { transaction, .. } => transaction.clone(),
             Request::Mempool { transaction, .. } => transaction.clone(),
         }
     }
 
+    /// The set of additional known unspent transaction outputs that's in this request.
     pub fn known_utxos(&self) -> Arc<HashMap<transparent::OutPoint, zs::Utxo>> {
         match self {
             Request::Block { known_utxos, .. } => known_utxos.clone(),
             Request::Mempool { known_utxos, .. } => known_utxos.clone(),
         }
     }
 
+    /// The network upgrade to consider for the verification.
+    ///
+    /// This is specified either explicitly in a request to verify a mempool transaction, or based
+    /// on the block height specified in a request to validate a transaction in a block.
```

Comment:
The mempool docs will be wrong as soon as we fix #1683, so let's just say this:
```suggestion
    /// This is based on the block height from the request, and the supplied `network`.
```